### PR TITLE
switch to new meson based testsuite runner

### PR DESCRIPTION
**Pull request by benoit-pierre**
- better isolation, with a separate process for each test file: crash isolation, avoid duplicated init errors (CRengine fonts, lua-ljsqlite3 init, …), separate data directory (no testsuite leftovers outside the install directory)
- faster thanks to parallelization
- better terminal output and full log as well as Junit XML test results (automatically, for each run)

On Circle CI, we can drop the parallelism, halving resources. On master, instead of 3 different runs (normal, coverage, and test results for timings), we can run the full suite one time with coverage, all the while reducing the build job wall-clock time: 2m 19s → 1m 13s.

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/12789)

### CODE REVIEW

File: spec/unit/filemanager_spec.lua [hunk 1]
```diff
@@ -1,21 +1,24 @@
 describe("FileManager module", function()
-    local FileManager, lfs, docsettings, UIManager, Screen, util
+    local DataStorage, FileManager, lfs, docsettings, UIManager, Screen, makePath, util
     setup(function()
         require("commonrequire")
+        -- G_reader_settings:saveSetting("document_metadata_folder", "doc")
```
Comment (Frenzie) [inline spec/unit/filemanager_spec.lua:5]: I'm not sure if I quite understand what the purpose of this line is?

**Review by Frenzie — APPROVED** [review]

**Comment by benoit-pierre** [comment]
```
No cache is found for key: 1-build-ccache-arch1-linux-amd64-6_85-iQMxO3NQF8CwxiZaIy899JkvY4VeFiI5sLcehUqFPAY=
Found a cache from build 17288 at 1-build-ccache-arch1-linux-amd64-6_85-
Size: 32 B
Cached paths:
  * /home/ko/.ccache
```
Somehow I doubt this cache is going to be much help… ;(

**Comment by benoit-pierre** [comment]
And indeed, an empty cache was saved…

https://app.circleci.com/pipelines/github/koreader/koreader/11706/workflows/4da444f8-b969-4bb9-b605-7798cd95c492/jobs/17288/parallel-runs/0/steps/0-110

**Comment by Frenzie** [comment]
Not found? Strange indeed. O.o

**Pull request merged**